Missing #include <typeinfo> in Kokkos_ExecPolicy.hpp

MWE:
```
#include <Kokkos_ExecPolicy.hpp>
```

Using clang-5.0 in the SIERRA environment results in the following error:
```
Kokkos_ExecPolicy.hpp:842:53: error: member access into incomplete type 'const std::type_info'
      default_name = std::string(typeid(FunctorType).name()) + "/" +
                                                    ^
/sierra/sntools/SDK/compilers/gcc/4.9.3-RHEL6/lib/gcc/x86_64-unknown-linux-gnu/4.9.3/../../../../include/c++/4.9.3/bits/exception_ptr.h:47:9: note: forward declaration of 'std::type_info'
  class type_info;
        ^
```

--- a/core/src/Kokkos_ExecPolicy.hpp
+++ b/core/src/Kokkos_ExecPolicy.hpp
@@ -52,6 +52,9 @@
 #include <impl/Kokkos_AnalyzePolicy.hpp>
 #include <Kokkos_Concepts.hpp>
 #include <iostream>
+#if defined(KOKKOS_ENABLE_PROFILING)
+#include <typeinfo>
+#endif // KOKKOS_ENABLE_PROFILING
 
 //----------------------------------------------------------------------------
 

